examples/general/incorrect_matches_operation/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+#![feature(rustc_private)]
+#![warn(unused_extern_crates)]
+
+extern crate rustc_ast;
+extern crate rustc_span;
+
+use clippy_utils::{diagnostics::span_lint, sym};
+use if_chain::if_chain;
+use rustc_ast::{
+    token::TokenKind,
+    tokenstream::TokenTree,
+    Expr, ExprKind, BinOpKind, MacCall, ptr::P,
+
+};
+use rustc_lint::{EarlyContext, EarlyLintPass};
+
+
+dylint_linting::impl_pre_expansion_lint! {
```

Comment:
Since you don't need to store anything, you should be able to use [`declare_pre_expansion_lint`](https://docs.rs/dylint_linting/2.4.3/dylint_linting/macro.declare_pre_expansion_lint.html) and avoid having to declare `IncorrectMatchesOperation` manually.

Reply:
Your point is very well taken.